bin/deploy.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+set -e
+
+REGION=us-east-2
+ECR_REGISTRY=377928551571.dkr.ecr.us-east-2.amazonaws.com
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --environment) ENVIRONMENT="$2"; shift ;;
+        --resource) RESOURCE="$2"; shift ;;
+        *) echo "Unknown parameter: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+# Check if required arguments are provided
+if [ -z "$ENVIRONMENT" ] || [ -z "$RESOURCE" ]; then
+    echo "Usage: $0 --environment <env> --resource <resource>"
+    exit 1
+fi
+
+# Derive values from environment and resource
+CLUSTER_NAME="$ENVIRONMENT-$RESOURCE-cluster"
+SERVICE_NAME="$ENVIRONMENT-$RESOURCE-service"
+TASK_DEF_NAME="$ENVIRONMENT-$RESOURCE-task-def"
+ECR_REPO="$RESOURCE-$ENVIRONMENT"
+
+# Fetch the latest image URI from ECR
+ECR_URL="$ECR_REGISTRY/$ECR_REPO"
+IMAGE_URI=$(aws ecr describe-images --repository-name $ECR_REPO --region $REGION --query 'sort_by(imageDetails,& imagePushedAt)[-1].imageTags[0]' --output text)
```

Comment:
empirically, this has often given me the `:latest`-tagged image URI, as opposed to the `COMMIT_HASH`-tagged image URI. This is unsafe to put in the task definition as new tasks getting started will use whatever image is tagged w/ latest at launch time. I'd propose we just generate `COMMIT_HASH` as we do in `build_and_push.sh` and use that to construct the `FULL_IMAGE_URI`

Reply:
done